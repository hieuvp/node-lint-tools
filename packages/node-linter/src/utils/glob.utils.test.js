const { getPatternByExtension } = require('./glob.utils');

describe('getPatternByExtension', () => {
  ['string', true, undefined, null, 1].forEach(args => {
    it(`should throw an error because "${args}" is not an array`, () => {
      expect(() => getPatternByExtension(args)).toThrowErrorMatchingSnapshot();
    });
  });

  it('should throw an error if args is empty', () => {
    expect(() => getPatternByExtension([])).toThrowErrorMatchingSnapshot();
  });
});
