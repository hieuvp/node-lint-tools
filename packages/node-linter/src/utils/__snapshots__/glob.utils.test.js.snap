// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPatternByExtension should throw an error because "1" is not an array 1`] = `"Expected \\"Array\\", instead got \\"1: number\\""`;

exports[`getPatternByExtension should throw an error because "null" is not an array 1`] = `"Expected \\"Array\\", instead got \\"null: object\\""`;

exports[`getPatternByExtension should throw an error because "string" is not an array 1`] = `"Expected \\"Array\\", instead got \\"string: string\\""`;

exports[`getPatternByExtension should throw an error because "true" is not an array 1`] = `"Expected \\"Array\\", instead got \\"true: boolean\\""`;

exports[`getPatternByExtension should throw an error because "undefined" is not an array 1`] = `"Expected \\"Array\\", instead got \\"undefined: undefined\\""`;

exports[`getPatternByExtension should throw an error if args is empty 1`] = `"Args \\"[]\\" must have at least one element"`;
